docs/practical-examples/forms.md

```diff
@@ -0,0 +1,190 @@
+---
+layout: default
+title: Forms
+parent: Practical Examples
+---
+
+# Forms
+
+## Table of contents
+
+1. [Before you start](#before-you-start)
+2. [Using form helpers inside a `ViewComponent`](#using-form-helpers-inside-a-view-component)
+3. [Creating your own input elements](#creating-your-own-input-elements)
+    1. [Writing input elements in vanilla html](#writing-input-elements-in-vanilla-html)
+4. [Gotchas and known issues](#gotchas-and-known-issues)
+
+## Before you start
+
+It's important to understand the building blocks of forms in Ruby on Rails before attempting to use `ViewComponent` to enhance them.
+
+Please refer to the official documentation and Rails Guides to get up to speed.
+
+- `ActionView` form helpers [visit](https://guides.rubyonrails.org/form_helpers.html){:target="_blank"}
+- `ActionView::Helpers::FormBuilder` [visit](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html){:target="_blank"}
+
+After reading these, you should have a good understanding of:
+
+- The `form_with` helper used to create an HTML `<form>`
+- `FormBuilder` and the instance yielded to the `form_with` block
+- The different helper methods you can call on the `FormBuilder` instance to create input fields (e.g. `form.text_field`)
+- The conventions set by Rails to [name parameters](https://guides.rubyonrails.org/form_helpers.html#understanding-parameter-naming-conventions){:target="_blank"}
+
+## Using form helpers inside a `ViewComponent`
+
+If your component needs to render a form, you can use the `form_with` helpers just as you would in a Rails template or partial
+
+```ruby
+# app/components/comment_form_component.rb
+class CommentFormComponent < ApplicationComponent
+  def initialize(comment:)
+    @comment = comment
+  end
+  attr_reader :comment
+end
+```
+
+```erb
+<!-- app/components/comment_form_component.html.erb -->
+<%= form_with(model: comment) do |form| %>
+  <%= form.label :content %>
+  <%= form.text_area :content %>
+<% end %>
+```
+
+## Creating your own input elements
+
+As your application grows, you might want to create more complex input types comprised of labels, inputs, wrapping
+divs, error messages, hints etc. To do this, Rails allows you to overwrite or create new form helpers.
+
+We'll use the simple example presented above to demonstrate how to create an abstraction for a field with input and
+label.
+
+### Using `FormBuilder` methods
+
+The first level of abstraction you can use is to create a new `FormBuilder` for your application and create your new
+helper method called `text_area_with_label`. Note that this form builder inherits from
+`ActionView::Helpers::FormBuilder` which provides all default form builder methods for you.
```

Comment:
As a general note, I believe we should be letting Jekyll handle newlines, not adding them ourselves.